Improve docs to be more explanatory

avinal opened the pull request:
<!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to a pull request) and what changes you've made, we can triage your pull request to the best possible team for review.

See our [CONTRIBUTING.md](/main/CONTRIBUTING.md) for information how to contribute.

For changes to content in [site policy](https://github.com/github/docs/tree/main/content/github/site-policy), see the [CONTRIBUTING guide in the site-policy repo](https://github.com/github/site-policy/blob/main/CONTRIBUTING.md).

We cannot accept changes to our translated content right now. See the [contributing.md](/main/CONTRIBUTING.md#earth_asia-translations) for more information.

Thanks again!
-->

### Why:

Fixes #974 
<!-- 
- If there's an existing issue for your change, please link to it.
- If there's _not_ an existing issue, please open one first to make it more likely that this update will be accepted: https://github.com/github/docs/issues/new/choose. -->

### What's being changed:

* Added `{:copy}` snippet to all code snippets
* Improved Dockerfile snippet to make scripts executable [here](https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-docker-container-action#creating-a-dockerfile) 
```dockerfile
# Container image that runs your code
FROM alpine:3.10
# Copies your code file from your action repository to the filesystem path `/` of the container
COPY entrypoint.sh /entrypoint.sh
# Makes your code file executable
RUN chmod +x entrypoint.sh
# Code file to execute when the docker container starts up (`entrypoint.sh`)
ENTRYPOINT ["/entrypoint.sh"]
```

* Removed a step [here](https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-docker-container-action#writing-the-action-code) because it was unclear, this step is now covered by the previous change,

<!-- Share artifacts of the changes, be they code snippets, GIFs or screenshots; whatever shares the most context. -->

### Check off the following:
- [x] All of the tests are passing.
- [x] I have reviewed my changes in staging.
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

avinal committed:
Improve docs to be more explanatory

avinal committed:
corrected {:yaml} to {:copy}

avinal reviewed (commented):
corrected

janiceilene commented:
@avinal Thanks so much for opening a PR! I'll get this triaged for review 🌟

avinal commented:
Is there more to add to this pr?

peterchyke commented:
> Is there more to add to this pr?

[Azure](http://wap.ng.zain.com/172.18.254.5)

peterchyke commented:
Organizer

janiceilene commented:
There's nothing you need to add @avinal! Thanks for your patience as our small team works through all of the amazing contributions 🌟

avinal commented:
Thanks, @janiceilene for confirmation.

Chitnarong06 commented on content/actions/creating-actions/creating-a-docker-container-action.md:
```diff
@@ -38,7 +38,7 @@ Before you begin, you'll need to create a GitHub repository.
 
 1. From your terminal, change directories into your new repository.
 
```
+ unlock

medoahmammmm commented:
medo-664@hotmail.com

martin389 committed:
Merge branch 'main' into patch-1

martin389 commented:
Thanks for this PR! 🎉  I'm just checking this recommendation with our support folks.

martin389 commented on content/actions/creating-actions/creating-a-docker-container-action.md:
```diff
@@ -47,13 +47,16 @@ Before you begin, you'll need to create a GitHub repository.
 # Copies your code file from your action repository to the filesystem path `/` of the container
 COPY entrypoint.sh /entrypoint.sh
 
+# Makes your code file executable
+RUN chmod +x entrypoint.sh
+
```
@avinal 👋 
Would you be able to change this to the original `chmod +x entrypoint.sh` version? The feedback I've received is that this approach adds another layer to the container, which would affect pull performance.

avinal replied:
Yeah, this adds another layer to the container. But as much I have used Dockerized GitHub Action, the script will not run without permission, this is also evident from the issue comment https://github.com/github/docs/issues/974#issue-734169633 for which this PR was submitted. You may want to see into that matter again. 

I can suggest an improvement to this, we can combine this line with other commands using `&&` such that it gets executed without an extra layer.

martin389 replied:
@avinal - sorry about the delay here 🙇  It looks like the approach in the existing documentation is to run `chmod +x entrypoint.sh` separately: (step 2 here) https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-docker-container-action#writing-the-action-code. Was this step not working for you?

avinal replied:
@martin389 Sorry for the delay, In the step two it is mentioned that run this command to make your file executable, but it is not mentioned that where you are supposed to run `chmod +x entrypoint.sh`, is it our system or inside docker? And that is the point of whole confusion. Secondly I am unsure about if the file will remain executable inside docker too. 

Secondly I had to add this line before running the entrypoint.sh file(maybe because I didn't already made them executable, no body does !!) Please give me 1 hour and I will test the exact steps described in the docs.

martin389 replied:
Thanks @avinal - the intention is for `chmod +x entrypoint.sh` to run on the user's system, and not within Docker. I agree that this is something that could be clearer 👍 

I tested these steps yesterday on macOS and the script remained executable, but I'd be interested to know if it works for you too. Thanks for all your help here! 🎊

avinal replied:
Here are the test cases I did
- Not making them executable in system - NO(expected)
- Making them executable in system using `chmod +x entrypoint.sh`- NO(not expected)
- Making them executable in system using `git update-index --chmod=+x entrypoint.sh` - NO(not expected)
- Making the system same (both ubuntu:latest) - NO(not expected)
- Adding the specific line in Dockerfile - YES(as usual)

Surprisingly the example works but the exact copy doesn't. I am pretty confused now😅

martin389 replied:
That would be great, thank you! 🎊

martin389 commented:
Hi @avinal 👋 
So sorry about this, but we're having some unrelated backend issues with this PR. Would you mind re-raising these awesome changes as a new PR (without deleting the forked repository)? 🙇 

For the new PR, I've also included some feedback about `chmod +x entrypoint.sh` in a separate comment for you to consider.
Thank you!

avinal commented:
Hello @martin389, 
No worries, I will submit a PR again :) Since this PR is almost a month old, unknowingly I deleted the fork.  May I fork again and then redo the changes? If this doesn't create any problem. 

> For the new PR, I've also included some feedback about chmod +x entrypoint.sh in a separate comment for you to consider.

I have added my feedback on this, please resolve that issue and let me know what is decided. 
Thanks !
